### ISSUE TITLE
Enable x-go-name in parameters

### ISSUE THREAD

**Issue by maxatome** [go-swagger/go-swagger#2167]
https://github.com/go-swagger/go-swagger/blob/master/generator/templates/server/urlbuilder.gotmpl#L109-L114

What do you think of allowing `x-go-name` in parameters:
```
  {{ range .PathParams }}
  {{- $goName := index .Extensions "x-go-name" }}
  {{ if $goName }}{{ $goName }}{{ else }}{{ pascalize .ID }}{{ end }} {{.GoType}}
  {{- end }}
  {{ range .QueryParams }}
  {{- $goName := index .Extensions "x-go-name" }}
  {{ if $goName }}{{ $goName }}{{ else }}{{ pascalize .ID }}{{ end }} {{ if and (not .IsArray) .IsNullable }}*{{ end }}{{.GoType}}
  {{- end }}
```
Not really needed on `PathParams`, but to be consistent why not.

On `QueryParams` it allows to expose params like `string` for example (compilation error, as `String` method is defined too).

**Comment by maxatome** [comment]
Shorter version:
```
  {{ range .PathParams }}
  {{ or (index .Extensions "x-go-name") (pascalize .ID) }} {{.GoType}}
  {{- end }}
  {{ range .QueryParams }}
  {{ or (index .Extensions "x-go-name") (pascalize .ID) }} {{ if and (not .IsArray) .IsNullable }}*{{ end }}{{.GoType}}
  {{- end }}
```

**Comment by fredbi** [comment]
@maxatome how would this work as far as parameter structures and validation code are generated for client and server? could you please detail the intent in just changing the URL builder?

**Comment by maxatome** [comment]
@fredbi I didn't use client generation, that is why I forgot it, sorry.
Look at #2168, I have just added the client part in generator/templates/client/parameter.gotmpl

**Comment by fredbi** [comment]
I've looked at the PR. I think it is breaking the generator. Part of it is because of the existing mess in templates (switching from `.ID` to `.Name` without reason, btw this would deserve a fix too...). Part of this is that when following a template-based approach, you are compelled to figure out every impact in all templates (client, server, validator, urlbuilder). I would recommend to apply the naming change in `operation.go` instead, so it propagates to every templates.

**Comment by maxatome** [comment]
@fredbi merged #2168, thanks!
